typing/btype.ml

```diff
@@ -651,136 +639,23 @@ let rec extract_label_aux hd l = function
-      | Some _, Some _ ->
-          if ty.level < ty'.level then
-            (log_type ty'; Private_type_expr.set_desc ty' (Tvar name))
-      | None, None   -> ()
-      end
-  | _ -> ()
-  (* ; assert (check_memorized_abbrevs ()) *)
-  (*  ; check_expans [] ty' *)
-(* TODO: consider eliminating set_type_desc, replacing it with link types *)
-let set_type_desc ty td =
-  if td != ty.desc then begin
-    log_type ty;
-    Private_type_expr.set_desc ty td
-  end
-(* TODO: separate set_level into two specific functions: *)
-(*  set_lower_level and set_generic_level *)
- let set_level ty level =
-  if level <> ty.level then begin
-    if ty.id <= !last_snapshot then log_change (Clevel (ty, ty.level));
-    Private_type_expr.set_level ty level
-  end
-(* TODO: introduce a guard and rename it to set_higher_scope? *)
-let set_scope ty scope =
-  if scope <> ty.scope then begin
-    if ty.id <= !last_snapshot then log_change (Cscope (ty, ty.scope));
-    Private_type_expr.set_scope ty scope
-  end
-let set_univar rty ty =
-  log_change (Cuniv (rty, !rty)); rty := Some ty
-let set_name nm v =
-  log_change (Cname (nm, !nm)); nm := v
-let set_row_field e v =
-  log_change (Crow (e, !e)); e := Some v
-let set_kind rk k =
-  log_change (Ckind (rk, !rk)); rk := Some k
-let set_commu rc c =
-  log_change (Ccommu (rc, !rc)); rc := c
-
-let snapshot () =
-  let old = !last_snapshot in
-  last_snapshot := !new_id;
-  (!trail, old)
-
-let rec rev_log accu = function
-    Unchanged -> accu
-  | Invalid -> assert false
-  | Change (ch, next) ->
-      let d = !next in
-      next := Invalid;
-      rev_log (ch::accu) d
-
-let backtrack (changes, old) =
-  match !changes with
-    Unchanged -> last_snapshot := old
-  | Invalid -> failwith "Btype.backtrack"
-  | Change _ as change ->
-      cleanup_abbrev ();
-      let backlog = rev_log [] change in
-      List.iter undo_change backlog;
-      changes := Unchanged;
-      last_snapshot := old;
-      trail := changes
-
-let rec rev_compress_log log r =
-  match !r with
-    Unchanged | Invalid ->
-      log
-  | Change (Ccompress _, next) ->
-      rev_compress_log (r::log) next
-  | Change (_, next) ->
-      rev_compress_log log next
-
-let undo_compress (changes, _old) =
-  match !changes with
-    Unchanged
-  | Invalid -> ()
-  | Change _ ->
-      let log = rev_compress_log [] changes in
-      List.iter
-        (fun r -> match !r with
-          Change (Ccompress (ty, desc, d), next) when ty.desc == d ->
-            Private_type_expr.set_desc ty desc; r := !next
-        | _ -> ())
-        log
-
-(* Mark a type. *)
-
-let not_marked_node ty = ty.level >= lowest_level
+let not_marked_node ty = get_level ty >= lowest_level
     (* type nodes with negative levels are "marked" *)
 
-let flip_mark_node ty = Private_type_expr.set_level ty (pivot_level - ty.level)
-let logged_mark_node ty = set_level ty (pivot_level - ty.level)
+let flip_mark_node ty =
+  let ty = Transient_expr.repr ty in
```

Comment:
This looks like a bug. At the very least, it goes against the documentation of the function in the mli, which explicitly states "No [repr]'ing".

Reply:
The comments in the mli were forgotten when switching to `always_repr`.
Repr'ing is now the correct behavior everywhere (except in `Printtyp`).
Fixed in #10665 .